### PR TITLE
fix hardcoded node name by selecting the first node for shard always

### PR DESCRIPTION
**Pull request by reyreaud-l**
### What's being changed:

Currently querier has a harcoded node name of `weaviate-0` to select the node for a given shard. This doesn't work in practice with cluster where shards are not on weaviate-0.

To remediate that this PR updates the way we retrieve tenant information to communicate back the `belongsToNode` information back to the querier (follow up PR once we merge the updated tenant data version endpoint, to propagate the information)

This has the limitations that if the tenant is replicated to multiple shards we will only read one replication of it. So if a user writes with "ONE" or "QUORUM" there's no guarantee that we will ever
1. repair so that all replication of the shards are the same
2. the querier will ever pick that write

For these reasons, we will recommend using the querier with no replication factor. This has the drawback that writes would fail if a node is restarting/unavailable as there would be no replica to pick to continue with the write.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: exp/query/lsm_fetcher.go [hunk 1]
```diff
@@ -85,11 +77,11 @@ func NewLSMFetcherWithCache(basePath string, upstream LSMDownloader, cache LSMCa
 // 2. If cache is enabled and not found in cache, fetch from upstream.
 // 3. If cache is enabled and found in the cache but local version is outdated, fetch from the upstream and remove old version.
 // 4. if cache is enabled, found in the cache and local version is up to date, return from the cache.
-func (l *LSMFetcher) Fetch(ctx context.Context, collection, tenant string, version uint64) (*lsmkv.Store, string, error) {
-	// unique key for (collectin, tenant, version). this key make sure only one in-flight requests hitting upstream.
-	key := fmt.Sprintf("%s-%s-%d", collection, tenant, version)
+func (l *LSMFetcher) Fetch(ctx context.Context, node, collection, tenant string, version uint64) (*lsmkv.Store, string, error) {
+	// unique key for (collection, tenant, version). this key make sure only one in-flight requests hitting upstream.
+	key := fmt.Sprintf("%s-%s-%s-%d", node, collection, tenant, version)
```
Comment (nathanwilk7) [inline exp/query/lsm_fetcher.go:82]:
```suggestion
	// unique key for (node, collection, tenant, version). this key make sure only one in-flight requests hitting upstream.
	key := fmt.Sprintf("%s-%s-%s-%d", node, collection, tenant, version)
```
@reyreaud-l do you think including the node in the key is necessary? or should we just assume the node doesn't matter? i think we'll usually get the same node for a given collection/tenant, but i'm trying to think through what happens if that node changes. @kavirajk any thoughts?

**Review by nathanwilk7 — APPROVED** [review]
lgtm, non-blocking question about removing node from the fetch key

File: exp/query/api.go [hunk 2]
```diff
@@ -87,7 +90,15 @@ func NewAPI(
 
 // Search serves vector search over the offloaded tenant on object storage.
 func (a *API) Search(ctx context.Context, req *SearchRequest) (*SearchResponse, error) {
-	info, tenantVersion, err := a.schema.TenantStatus(ctx, req.Collection, req.Tenant)
+	// Expectations
```
Comment (kavirajk) [inline exp/query/api.go:93]: I think we can remove this comment. It's too implementation oriented I think. probably you got back when rebasing your branch.

File: exp/query/lsm_fetcher.go (hunk 1 above)
Comment (kavirajk) [reply exp/query/lsm_fetcher.go:82]: it totally depends what we need to do if the node changes with same version? we call upstream or no?

I mean given, the replicas can be diverged, still I would love to call upstream only if the version is changed. So basically I would prefer to keep the node out of the key. 

Also curious about the rationale why @reyreaud-l included it.

File: cluster/schema/schema.go [hunk 3]
```diff
@@ -414,9 +422,10 @@ func makeTenant(name, status string) models.Tenant {
...
 
 // MakeTenantWithDataVersion creates a tenant with the given name, status, and data version
-func MakeTenantWithDataVersion(name, status string, dataVersion int64) *models.TenantResponse {
+func MakeTenantWithDataVersion(name, status string, belongsToNodes []string, dataVersion int64) *models.TenantResponse {
 	return &models.TenantResponse{
-		Tenant:      makeTenant(name, status),
-		DataVersion: &dataVersion,
+		Tenant:         makeTenant(name, status),
+		DataVersion:    &dataVersion,
+		BelongsToNodes: belongsToNodes,
```
Comment (kavirajk) [inline cluster/schema/schema.go:429]: nit: `BelongsToNode` -> `OwnedNodes`. Feel bit more accurate to me.

**Review by kavirajk — COMMENTED** [review]
nice work. LGTM. left few minor comments. but no blocker.

**Review by kavirajk — APPROVED** [review]

**Review by nathanwilk7 — APPROVED** [review]
approved (node removed from fetch key)

File: cluster/schema/schema.go (hunk 3 above)
Comment (reyreaud-l) [reply cluster/schema/schema.go:429]: I'm reusing the same nomenclature that is used everywhere else in the sharding state. And here I would value consistency over different naming :)

**Review by nathanwilk7 — APPROVED** [review]
lgtm

**Review by nathanwilk7 — APPROVED** [review]
lgtm

**Pull request merged**